.github/workflows/main.yml

```
@@ -1,16 +1,79 @@
 # Run test file with command:
 #   act pull_request -e testdata/act/pull-request.json
 
-name: Test find flags
+name: Test and Generate Docs
+
 on: pull_request
 
 jobs:
-  find-feature-flags:
+  go-tests:
     runs-on: ubuntu-latest
-    name: Test Find Flags
     steps:
       - name: Checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18.x'
+
+      - name: Install datadog-ci
+        run: npm install -g @datadog/datadog-ci
+
+      - name: Set up Python
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b
+        with:
+          python-version: '3.x'
+
+      - name: Install pre-commit
+        run: |
+          python -m pip install --upgrade pip
+          pip install pre-commit
+
+      - name: Run pre-commit hooks
+        run: pre-commit run --all-files
+
+      - name: Set up Go
+        uses: actions/setup-go@41dfa10bad2bb2ae585af6ee5bb4d7d973ad74ed
+        with:
+          go-version-file: go.mod
+
+      - name: Install dependencies
+        run: go mod tidy
+
+      - name: Install gotestsum
+        run: go install gotest.tools/gotestsum@latest
+
+      - name: Run tests with gotestsum
+        run: |
+          mkdir -p ${{ github.workspace }}/artifacts
+          mkdir -p ${{ github.workspace }}/reports
+
+          GONFALON_MODE=test \
+          gotestsum --packages="./..." \
+            --junitfile ${{ github.workspace }}/reports/go-test_go_tests.xml \
+            --jsonfile ${{ github.workspace }}/artifacts/go-test_go_tests.json \
+            --rerun-fails=2 \
+            --rerun-fails-max-failures=10 \
+            --rerun-fails-report ${{ github.workspace }}/artifacts/rerun_tests_go_tests.txt \
+            -- -tags=launchdarkly_easyjson -p=1
+
+      - name: Publish JUnit Tests
+        uses: ./.github/actions/publish-junit
+        env:
+          DD_API_KEY: ${{ secrets.DATADOG_API_KEY }}
+        with:
+          files: ${{ github.workspace }}/reports/go-test_go_tests.xml
+          name: Go Tests
```

probably better to call this `find-code-references-in-pull-request` or something. This will show up as the "service" in `CI Test Visibility`
```suggestion
          name: find-code-references-in-pull-request
```

---

updated the name